src/request/params/query_parser.rs

```diff
@@ -1,103 +1,81 @@
 use percent_encoding::percent_decode_str;
-use std::{iter::Peekable, str::CharIndices};
-
-pub fn parse_query_params(input: &str) -> Vec<(String, (usize, usize))> {
-    let mut query_params = Vec::with_capacity(16);
-    let mut input = QueryParserInput::new(input);
-
-    while let Some(query_param) = parse_query_param(&mut input) {
-        query_params.push(query_param);
-    }
+use std::borrow::Cow;
+use std::iter::Peekable;
+use std::str::CharIndices;
+
```

Comment:
This is a bit of a sloppy implementation of a parser but it's functional and sufficient for the weekend. I'll fix this when I come back to work on Monday.